src/components/NcAppSidebar/NcAppSidebar.vue

```diff
@@ -1135,7 +1147,16 @@ export default {
 		 * @public
 		 */
 		focus() {
-			(this.$refs.header ?? this.$refs.toggle)?.focus()
+			if (!this.open && !this.noToggle) {
+				this.$refs.toggle.$el.focus()
+				return
+			}
+
+			try {
+				document.getElementById(this.headerId).focus()
+			} catch (e) {
```

Comment:
Note: alternative for providing ID is to provide a (template) ref.

Reply:
@Antreesy By "provide" I meant Vue provide to the NcAppSidebarHeader. Then the header can inject it and bind as a template ref.

ID works, this component is single use and has ID anyway, but ref would be slightly more vue way

Reply:
Oh.